### ISSUE TITLE
Upgrade onboard to use Wallet-Connect V2

### ISSUE THREAD

**Issue by L03TJ3** [GoodDollar/GoodProtocolUI#325]
WalletConnect V1 will be deprecated in a few months' time (June 28 2023). 
gd-wallet has already been updated to v2, it might be good to do the same on ProtocolUI

- [x] - Make sure latest onboard package(s) are used
- [x] - Update to version v2 https://onboard.blocknative.com/docs/wallets/walletconnect#install
- [x] - Verify if there are changes needed for custom WC implementation in sdk-v2/onboard 
- [x] - wallet must be defined (error from web3-onboard/core) (test with latest alpha release)
- [x] - Wait for the official onboard package to be released
- [x] - wallet must be defined (error from web3-onboard/core) https://app.zenhub.com/workspaces/protocol-ui-team-61559a6a33275e00141f8ee4/issues/gh/gooddollar/goodprotocolui/325#issuecomment-1534306583

- [x] add svg icon for valora
- [x] add png/webp icons to wc2 config walletIcons array
- [ ] #416
- [x] verify that valora is working
   - its set to use v2
   - verify deeplink triggers on click (android/ios)
   
 Active bugs:
- [x] - [Missing website name](https://github.com/GoodDollar/GoodWeb3-Mono/issues/83)
- [ ] - Chain id not updated in wallet https://github.com/GoodDollar/GoodDAPP/issues/3994
- [x] - Wallet does not update ui after disconnecting https://github.com/GoodDollar/GoodDAPP/issues/3994
- [x] -  disconnecting on GoodDAPP doesn't reload and breaks ui
- [x] - UI doesn't update after claiming (fuse)

How to test @vldkh  for wallet-connect V2:
[GoodDollar Wallet]: Both Desktop and android. Scanning QR / Copy to clipboard / Deeplink
- connecting
- switching
- doing tx
- disconnecting
[Valora Wallet]: Only has mobile apps (android/ios)
Scan QR (desktop) / DeepLink on mobile
- connecting
- doing tx (maybe try FV on dev)
- disconnecting

**Comment by L03TJ3** [comment]
On hold until at least metamask has updated to support v2 as well
Est: end of Q1 2023

**Comment by sirpy** [comment]
@L03TJ3 why are the tasks checkmarked? are they done? is there a PR for this?

**Comment by L03TJ3** [comment]
I did it locally, but now realize I have not pushed anything
I shall create a PR for it, either later today or tomorrow morning

**Comment by L03TJ3** [comment]
Yes it has a fallback, i'll explain why its still testing/draft in the pr or slack

**Comment by L03TJ3** [comment]
Connecting works (it seems)
Current bug:
- [ ] - wallet must be defined (error from web3-onboard/core)

Steps to reproduce:
1. go to localhost (or latest PR deployment)
2. go to dev.gooddollar.org
3. on gooddapp, click on connect with wallet-connect + copy link
4. go to wallet, paste
5. try and switch
6. below error occurs

![image](https://user-images.githubusercontent.com/6606028/236152813-76a952d4-f6ba-4a39-a445-f860532d8f93.png)

**Comment by L03TJ3** [comment]
Added issue on their repo: https://github.com/blocknative/web3-onboard/issues/1677

**Comment by johnsmith-gooddollar** [comment]
@L03TJ3 please update checkboxes. and new issues should also be added/duplicated at the task description

**Comment by sirpy** [comment]
@L03TJ3 updated the description

**Comment by L03TJ3** [comment]
TestCases for wallet-connect V2:
[GoodDollar Wallet]: Both Desktop and android. Scanning QR / Copy to clipboard / Deeplink
- connecting
- switching
- doing tx
- disconnecting
[Valora Wallet]: Only has mobile apps (android/ios)
Scan QR (desktop) / DeepLink on mobile
- connecting
- doing tx (maybe try FV on dev)
- disconnecting

**Comment by L03TJ3** [comment]
Found bugs:
- [ ] - Missing website name
- [ ] - Chain id not updated in wallet
- [ ] - Wallet does not update ui after disconnecting
- [ ] - Slow connection breaks ui on GoodDAPP
- [ ] - Wallet-connect disconnect no longer work
- [ ] - UI doesn't update after claiming (fuse)

**Comment by sirpy** [comment]
@L03TJ3 do we have issues open for all the bugs mentioned?
Is there a v1 fallback? how does it work?

**Comment by L03TJ3** [comment]
I see I commented before that there is a fallback. Not sure what I meant but no there is not
either you initialize a v2 wc module, or a v1
but this needs to be decided beforehand since there is not a switch option

For now, the v1 fallback would apply to the default wallet-connect button, which still uses the v1 module

**Comment by L03TJ3** [comment]
The issue has been updated with the final bug ticket created
all known bugs are now accounted for

**Comment by vldkhh** [comment]
verified on prod
